### PR TITLE
gRIBI tests to explicitly specify persistence mode

### PR DESCRIPTION
**Pull request by liulk**

### CODE REVIEW

File: feature/gribi/otg_tests/ordering_ack_test/ordering_ack_test.go [hunk 1]
```diff
@@ -494,44 +492,61 @@ func TestOrderingACK(t *testing.T) {
...
-				n.WithFIBACK()
-				wantInstalled = fluent.InstalledInFIB
-			}
-
-			c.Start(ctx, t)
-			defer c.Stop(t)
-			c.StartSending(ctx, t)
-			if err := awaitTimeout(ctx, c, t); err != nil {
-				t.Fatalf("Await got error during session negotiation: %v", err)
+	for _, persist := range []string{"PRESERVE", "DELETE"} {
```
Comment (xw-g) [inline feature/gribi/otg_tests/ordering_ack_test/ordering_ack_test.go:495]: Same comments as above.

**Review by xw-g — COMMENTED** [review]
Thanks for the PR. Only some minor comments/questions.

File: feature/gribi/ate_tests/ack_in_presence_other_routes/README.md [hunk 2]
```diff
@@ -12,11 +12,9 @@ Ensure that ACKs are received in the presence of other routes.
 *   Configure static routes on the DUT for 203.0.113.0/24 pointing to ATE
     port-2. Ensure that the static route is installed in the DUT.
 
-*   Establish gRIBI client to the DUT.
-
-*   Connect gRIBI client to DUT specifying `SINGLE_PRIMARY` client redundancy in
-    the SessionParameters request, and `election_id` 10. Ensure that no error is
-    reported from the gRIBI server.
+*   Connect gRIBI client to DUT specifying persistence mode `PRESERVE`,
```
[the remark this answers is not in the document]
Comment (xw-g) [reply feature/gribi/ate_tests/ack_in_presence_other_routes/README.md:15]: nvm. I see it's already in persistent mode.

File: feature/gribi/ate_tests/base_leader_election_test/README.md [hunk 3]
```diff
@@ -12,12 +12,12 @@ Validate Election ID is accepted from a gRIBI client.
 *   Establish two gRIBI clients to the DUT (referred to as `gRIBI-A` and
     `gRIBI-B`).
 
-*   Connect `gRIBI-A` to DUT specifying `SINGLE_PRIMARY` client redundancy in
-    the SessionParameters request, and `election_id` 10. Ensure that no error is
-    reported from the gRIBI server.
+*   Connect `gRIBI-A` to DUT specifying `PRESERVE` persistent mode,
```
[the remark this answers is not in the document]
Comment (xw-g) [reply feature/gribi/ate_tests/base_leader_election_test/README.md:15]: nvm. I see it's already in persistent mode.

File: feature/gribi/otg_tests/ack_in_presence_other_routes/README.md [hunk 4]
```diff
@@ -12,11 +12,9 @@ Ensure that ACKs are received in the presence of other routes.
 *   Configure static routes on the DUT for 203.0.113.0/24 pointing to ATE
     port-2. Ensure that the static route is installed in the DUT.
 
-*   Establish gRIBI client to the DUT.
-
-*   Connect gRIBI client to DUT specifying `SINGLE_PRIMARY` client redundancy in
-    the SessionParameters request, and `election_id` 10. Ensure that no error is
-    reported from the gRIBI server.
+*   Connect gRIBI client to DUT specifying persistence mode `PRESERVE`,
```
[the remark this answers is not in the document]
Comment (xw-g) [reply feature/gribi/otg_tests/ack_in_presence_other_routes/README.md:15]: nvm. I see it's already in persistent mode.

File: feature/gribi/otg_tests/base_leader_election_test/README.md [hunk 5]
```diff
@@ -12,12 +12,12 @@ Validate Election ID is accepted from a gRIBI client.
 *   Establish two gRIBI clients to the DUT (referred to as `gRIBI-A` and
     `gRIBI-B`).
 
-*   Connect `gRIBI-A` to DUT specifying `SINGLE_PRIMARY` client redundancy in
-    the SessionParameters request, and `election_id` 10. Ensure that no error is
-    reported from the gRIBI server.
+*   Connect `gRIBI-A` to DUT specifying `PRESERVE` persistent mode,
```
[the remark this answers is not in the document]
Comment (xw-g) [reply feature/gribi/otg_tests/base_leader_election_test/README.md:15]: nvm. I see it's already in persistent mode.

File: feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 6]
```diff
@@ -430,38 +431,61 @@ func TestRecursiveIPv4Entries(t *testing.T) {
...
-
-			c.Connection().WithStub(gribic).WithInitialElectionID(10, 0).
-				WithRedundancyMode(fluent.ElectedPrimaryClient)
-
-			c.Start(context.Background(), t)
-			defer c.Stop(t)
-			c.StartSending(context.Background(), t)
-			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
-				t.Fatalf("Await got error during session negotiation for c: %v", err)
+	for _, persist := range []string{"PRESERVE", "DELETE"} {
```
[the remark this answers is not in the document]
Comment (liulk) [reply feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go]: TE-4.2 is the case that explicitly covers both `PRESERVE` and `DELETE` but it only tests the non-recursive IPv4/NHG/NH. This is the main reason why we're doing `PRESERVE` + optional `DELETE` for some test plans so can have better coverage.

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 7]
```diff
@@ -466,38 +467,61 @@ func TestRecursiveIPv4Entries(t *testing.T) {
...
-
-			c.Connection().WithStub(gribic).WithInitialElectionID(10, 0).
-				WithRedundancyMode(fluent.ElectedPrimaryClient)
-
-			c.Start(context.Background(), t)
-			defer c.Stop(t)
-			c.StartSending(context.Background(), t)
-			if err := awaitTimeout(ctx, c, t, time.Minute); err != nil {
-				t.Fatalf("Await got error during session negotiation for c: %v", err)
+	for _, persist := range []string{"PRESERVE", "DELETE"} {
```
[the remark this answers is not in the document]
Comment (liulk) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go]: Done.

File: feature/gribi/otg_tests/ordering_ack_test/ordering_ack_test.go (hunk 1 above)
Comment (liulk) [reply feature/gribi/otg_tests/ordering_ack_test/ordering_ack_test.go]: Done.

**Review by xw-g — APPROVED** [review]

**Pull request merged**